### DISCUSSION TITLE
Override decline terms and conditions

### DISCUSSION THREAD

**Discussion by hjroundi** [keycloak/keycloak#20492] [Q&A]
Hello, 

I am using keycloak 15.1.1 on java and xwiki project. When I update my terms and conditions and the user declines this update, he is redirected to the page: https://wiki-dev.companyname.eu/bin/login/XWiki/XWikiLogin?xredirect=% 2Fbin%2Fview%2FMain%2F
  but since the user is not logged in this returns a page with HTTP Status 400 – Bad Request. I would therefore like to be able to override the initial behavior of the EventListener by adding a redirect to a URI rather than keeping the current state. I tried to modify the value of "redirect_uri" of the event but it doesn't change much, it looks like keycloak is forcing the redirect defined initially.

Event Type : CUSTOM_REQUIRED_ACTION_ERROR
Error : rejected_by_user

Do you have a solution for this? If you need more information, I am available to answer you.

Thank you so much.

**Comment by jonkoops** [comment]
I am not able to answer your question, but I should warn you that it is very unlikely you will get an answer to your question running an outdated version of Keycloak. I would highly recommend you update to the latest version, as this issue might have already been resolved.

Aside from that, running outdated versions of Keycloak is a serious security concern.

**Reply by hjroundi** [reply]
Thank you for your reply.

Actually I don't have the power to update the version now. I have to work with this version and find a solution to bypass this issue.
